### PR TITLE
Tcp server stream 

### PR DESCRIPTION
**Pull request by lefendie**
TCP Server stream defines the connection between the client and the server.

### CODE REVIEW

File: includes/TCPStream.hpp [hunk 1]
```diff
@@ -17,27 +17,30 @@
...
-		void setEvent(const int event);
+		void setEvent(const uint32_t event);
 		void setMessage(const std::string &message);
 
 		int getFd() const;
-		sockaddr_in getData() const;
-		int getEvent() const;
+		IPAddress getData() const;
+		uint32_t getEvent() const;
 		std::string getMessage() const;
```
Comment (savosaicic) [inline includes/TCPStream.hpp]: return a const ref instead (rtfm)

File: includes/TCPStream.hpp [hunk 2]
```diff
@@ -17,27 +17,30 @@
...
-		int _event;
+		IPAddress _data;
+		uint32_t _event;
 		std::string _message;
+		int _port;
 	
 	public:
 		TCPStream();
-		TCPStream(int fd, sockaddr_in data, int event);
+		TCPStream(int fd, IPAddress data, uint32_t event);
```
Comment (savosaicic) [inline includes/TCPStream.hpp]: const IPAdress& instead of a copy

File: includes/TCPStream.hpp [hunk 3]
```diff
@@ -17,27 +17,30 @@
...
 		TCPStream(const TCPStream &src);
 		
-		void setEvent(const int event);
+		void setEvent(const uint32_t event);
 		void setMessage(const std::string &message);
 
 		int getFd() const;
-		sockaddr_in getData() const;
-		int getEvent() const;
+		IPAddress getData() const;
```
Comment (savosaicic) [inline includes/TCPStream.hpp]: return a const ref instead (rtfm)

**Review by savosaicic — CHANGES_REQUESTED** [review]

File: includes/TCPStream.hpp (hunk 2 above)
Comment (lefendie) [reply includes/TCPStream.hpp]: On the way !!!

File: includes/TCPStream.hpp (hunk 3 above)
Comment (lefendie) [reply includes/TCPStream.hpp]: okokok

**Review by savosaicic — APPROVED** [review]
Quel beau code !

**Pull request merged**